Aplib.Tests/Desire/GoalStructureTests.cs

```diff
@@ -53,49 +46,43 @@ public void FirstOfGoalStructure_WhenDisposing_ShouldDisposeChildren()
     [Fact]
     public void FirstOfGoalStructure_WhenFinished_ShouldEarlyExit()
     {
-        Mock<IGoalStructure<BeliefSet>> goalStructure1 = new();
+        Mock<IGoalStructure<IBeliefSet>> goalStructure1 = new();
         goalStructure1.SetupGet(g => g.Status).Returns(CompletionStatus.Success);
 
-        Mock<IGoalStructure<BeliefSet>> goalStructure2 = new();
+        Mock<IGoalStructure<IBeliefSet>> goalStructure2 = new();
         goalStructure2.SetupGet(g => g.Status).Returns(CompletionStatus.Success);
 
-        BeliefSet beliefSet = Mock.Of<BeliefSet>();
-        FirstOfGoalStructure<BeliefSet> firstOfGoalStructure = new(new List<IGoalStructure<BeliefSet>>
-        {
-            goalStructure1.Object, goalStructure2.Object
-        });
+        IBeliefSet beliefSet = Mock.Of<IBeliefSet>();
+        FirstOfGoalStructure<IBeliefSet> firstOfGoalStructure = new([goalStructure1.Object, goalStructure2.Object]);
 
         // Act
         firstOfGoalStructure.UpdateStatus(beliefSet);
         firstOfGoalStructure.UpdateStatus(beliefSet);
 
         // Assert
         firstOfGoalStructure.Status.Should().Be(CompletionStatus.Success);
-        goalStructure1.Verify(x => x.UpdateStatus(It.IsAny<BeliefSet>()), Times.Once);
-        goalStructure2.Verify(x => x.UpdateStatus(It.IsAny<BeliefSet>()), Times.Never);
+        goalStructure1.Verify(x => x.UpdateStatus(It.IsAny<IBeliefSet>()), Times.Once);
+        goalStructure2.Verify(x => x.UpdateStatus(It.IsAny<IBeliefSet>()), Times.Never);
     }
 
     [Fact]
     public void FirstOfGoalStructure_WhenFirstGoalIsFailure_ShouldReturnSecondGoal()
     {
         // Arrange
-        Mock<IGoalStructure<BeliefSet>> goalStructure1 = new();
+        Mock<IGoalStructure<IBeliefSet>> goalStructure1 = new();
         goalStructure1.SetupGet(g => g.Status).Returns(CompletionStatus.Failure);
-        Mock<IGoalStructure<BeliefSet>> goalStructure2 = new();
-        IGoal goal = Mock.Of<IGoal>();
+        Mock<IGoalStructure<IBeliefSet>> goalStructure2 = new();
+        IGoal<IBeliefSet> goal = Mock.Of<IGoal<IBeliefSet>>();
         goalStructure2
-            .Setup(g => g.GetCurrentGoal(It.IsAny<BeliefSet>()))
+            .Setup(g => g.GetCurrentGoal(It.IsAny<IBeliefSet>()))
             .Returns(goal);
 
-        BeliefSet beliefSet = Mock.Of<BeliefSet>();
-        FirstOfGoalStructure<BeliefSet> firstOfGoalStructure = new(new List<IGoalStructure<BeliefSet>>
-        {
-            goalStructure1.Object, goalStructure2.Object
-        });
+        IBeliefSet beliefSet = Mock.Of<IBeliefSet>();
+        FirstOfGoalStructure<IBeliefSet> firstOfGoalStructure = new([goalStructure1.Object, goalStructure2.Object]);
```

Comment:
CallBase has to be set here in the constructor, otherwise we won't get any coverage for dispose. Please add this.